pylint/checkers/base.py

```diff
@@ -1882,6 +1902,28 @@ def _raise_name_warning(self, node, node_type, name, confidence):
         self.add_message("invalid-name", node=node, args=args, confidence=confidence)
         self.stats["badname_" + node_type] += 1
 
+    def _name_valid_due_to_whitelist(self, name: str) -> bool:
+        good_names_patterns = self.config.good_names_rgxs
+        matches = [
+            match
+            for match in (pattern.search(name) for pattern in good_names_patterns)
+            if match is not None
+        ]
+        return len(matches) > 0 or name in self.config.good_names
```

Comment:
Can we flip these around? It's faster to check against `good_names` than to search against some regexes.

Reply:
done